pkg/render/gateway_api.go

```diff
@@ -647,6 +658,71 @@ func (pr *gatewayAPIImplementationComponent) envoyProxyConfig(defaultName string
 			envoyProxy.Spec.Provider.Kubernetes.EnvoyDeployment.Container = &envoyapi.KubernetesContainerSpec{}
 		}
 		envoyProxy.Spec.Provider.Kubernetes.EnvoyDeployment.Container.Image = &pr.envoyProxyImage
+
+		if pr.cfg.Installation.Variant == operatorv1.TigeraSecureEnterprise {
+			// Update the envoy proxy deployment to include the WAF HTTP filter
+
+			// Update Pod volumes
+			if envoyProxy.Spec.Provider.Kubernetes.EnvoyDeployment.Pod.Volumes == nil {
+				envoyProxy.Spec.Provider.Kubernetes.EnvoyDeployment.Pod.Volumes = []corev1.Volume{
+					{
+						VolumeSource: corev1.VolumeSource{
+							HostPath: &corev1.HostPathVolumeSource{
+								Path: "/var/log/calico",
+								Type: ptr.ToPtr[corev1.HostPathType](corev1.HostPathDirectoryOrCreate),
+							},
+						},
+						Name: "var-log-calico",
+					},
+					{
+						VolumeSource: corev1.VolumeSource{
+							EmptyDir: &corev1.EmptyDirVolumeSource{},
+						},
+						Name: "waf-http-filter",
+					},
+				}
+			}
+
+			// Add the init container to the deployment
+			if envoyProxy.Spec.Provider.Kubernetes.EnvoyDeployment.InitContainers == nil {
```

Comment:
What scenarios can the InitContainers be non-nil? And if they are non-nil, do we not want to insert this init container into them?